### PR TITLE
Pass through stream

### PR DESCRIPTION
**Pull request by alvestrand**
This is a demo of processing a video stream in a Worker.

### CODE REVIEW

File: src/content/capture/worker-process/js/main.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+(async () => {
+  inputStream = await navigator.mediaDevices.getUserMedia({video: true});
+  source.srcObject = inputStream;
+  transformStream = new TransformStream();
+  writer = transformStream.writable.getWriter();
+
+  const myWorker = new Worker('js/worker.js');
+  myWorker.onmessage = function(e) {
+    console.log('Message from worker: ' + e.data[0]);
```
Comment (guidou) [inline src/content/capture/worker-process/js/main.js]: Do you want to keep the console.log() messages?

File: src/content/capture/worker-process/js/worker.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+*  tree.
+*/
+
+'use strict;'
+
+onmessage = function(e) {
+  console.log('got a message');
+  let command = e.data[0];
+  let inputStream = e.data[1];
+  console.log('command is ' + command);
```
Comment (guidou) [inline src/content/capture/worker-process/js/worker.js]: ditto

File: src/content/capture/worker-process/js/main.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    // Start the flow of data.
+    readData();
+    window.requestAnimationFrame(loop);
+  }
+  myWorker.postMessage(['stream', transformStream.readable],
+		       [ transformStream.readable ]);
+
+  source.play();
+  result.play();
+
```
Comment (guidou) [inline src/content/capture/worker-process/js/main.js]: remove blank line

File: src/content/capture/worker-process/js/main.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  const myWorker = new Worker('js/worker.js');
+  myWorker.onmessage = function(e) {
+    console.log('Message from worker: ' + e.data[0]);
+    console.log(e.data[1]);
+    reader = e.data[1].getReader();
+    // Start the flow of data.
+    readData();
+    window.requestAnimationFrame(loop);
+  }
+  myWorker.postMessage(['stream', transformStream.readable],
```
Comment (guidou) [inline src/content/capture/worker-process/js/main.js:67]: why is transformStream.readable passed twice?

Comment (alvestrand) [reply src/content/capture/worker-process/js/main.js:67]: postMessage() has two arguments, one lists what to pass, and one lists what objects to neuter in the calling context. It doesn't make sense to have the readable stream present in both contexts, so we have to neuter it where it's sent.

Comment (guidou) [reply src/content/capture/worker-process/js/main.js:67]: thanks for the clarification

File: src/content/capture/worker-process/js/main.js (hunk 1 above)
Comment (alvestrand) [reply src/content/capture/worker-process/js/main.js]: No. removed.

File: src/content/capture/worker-process/js/main.js (hunk 3 above)
Comment (alvestrand) [reply src/content/capture/worker-process/js/main.js]: Done.

File: src/content/capture/worker-process/js/worker.js (hunk 2 above)
Comment (alvestrand) [reply src/content/capture/worker-process/js/worker.js]: Done.

**Review by alvestrand — COMMENTED** [review]
Addressed. Thanks!

**Review by guidou — APPROVED** [review]
lgtm

File: src/content/capture/worker-process/js/worker.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+*
+*  Use of this source code is governed by a BSD-style license
+*  that can be found in the LICENSE file in the root of the source
+*  tree.
+*/
+
+'use strict;'
+
+onmessage = function(e) {
+  console.log('got a message');
```
Comment (guidou) [inline src/content/capture/worker-process/js/worker.js]: there is still a console.log statement here

Comment (alvestrand) [reply src/content/capture/worker-process/js/worker.js]: Oops. Sorry.

**Pull request merged**